app/storybook/.storybook/preview.tsx

```diff
@@ -16,10 +18,62 @@ const GlobalStyles = () => (
         /* @TODO: find a way to customize it without !important dominance */
         color: #2e90fa !important;
       }
+      a.showCodeLink {
+        display: block;
+        margin-top: 20px;
+        cursor: pointer;
+        font-size: 0.9em;
+        color: #2e90fa;
+      }
     `}
   />
 )
 
+const withSource = (StoryFn, context) => {
+  const [showSource, setShowSource] = React.useState(false)
+
+  let source
+  try {
+    const docSource = context.parameters.docs.source.originalSource
+
+    // Parse render from the source string
+    const lines = docSource.split('\n')
+    const renderStr = 'render: args =>'
+    const renderStartIndex = lines.findIndex((line) => line.indexOf(renderStr) > -1)
+    const renderEndIndex = lines.findIndex(
+      (line, index) =>
+        index > renderStartIndex && (line === '}' || (line.substr(0, 2) === '  ' && line.charAt(2) !== ' '))
```

Comment:
I believe, there can be some elegant regex solution here. Not a pro at the thing :)